java-api-core/src/com/kinvey/java/AbstractClient.java

```diff
@@ -560,15 +560,15 @@ public SyncManager getSyncManager(){
         return new SyncManager(getSyncCacheManager());
     }
 
-    public Class getUserClass(){
-        return this.userModelClass;
+    public Class<T> getUserClass(){
+        return this.userModelClass != null ? this.userModelClass : BaseUser.class;
```

Comment:
Why have this logic here instead of defaulting `userModelClass` to `BaseUser.class`?

Reply:
It was done because `getUserClass` must return `Class<T>` type, but `BaseUser.class` should be cast to `Class<T>` before. I think I can return old implementation for `getUserClass` and add casting `userModelClass = (Class<T>) BaseUser.class`. It will be more expectable for users. Updated.